packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+
+const SearchBoxIcon = React.forwardRef((props, ref) => {
+  const { size, className, children, ...rest } = props;
+  const { size: sizeContext } = useSafeContext(SearchBoxContext);
+
+  return (
+    <Icon
+      aria-hidden
+      className={cx('iui-svg-icon', 'iui-search-icon', className)}
+      size={size ?? sizeContext}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', IconProps>;
+
+// ----------------------------------------------------------------------------
+
+export type SearchBoxInputProps = {
+  label?: string;
+} & React.ComponentProps<'input'>;
+
+const SearchBoxInput = React.forwardRef(
+  (props: SearchBoxInputProps, ref: React.Ref<HTMLInputElement>) => {
+    const {
+      className,
+      label = 'Search input',
+      id: idProp,
+      disabled: disabledProp,
+      ...rest
+    } = props;
+
+    const { inputId, setInputId, disabled, inputRef } =
+      useSafeContext(SearchBoxContext);
+
+    if (idProp && idProp !== inputId) {
+      setInputId(idProp);
+    }
+
+    return (
+      <input
+        id={idProp ?? inputId}
+        aria-label={label}
+        ref={useMergedRefs(ref, inputRef)}
+        type='text'
+        className={cx('iui-search-input', className)}
+        disabled={disabledProp ?? disabled}
+        {...rest}
+      />
+    );
+  },
+);
+
+// ----------------------------------------------------------------------------
+
+/**
+ * SearchBox.Button component to add to your SearchBox.
+ */
+export type SearchBoxButtonProps = PolymorphicComponentProps<
+  'button',
+  IconButtonProps
+>;
+
+const SearchBoxButton = React.forwardRef((props, ref) => {
+  const { children, size, disabled: disabledProp, ...rest } = props;
+  const { size: sizeContext, disabled } = useSafeContext(SearchBoxContext);
+
+  return (
+    <IconButton
+      styleType='borderless'
+      size={size ?? sizeContext}
+      ref={ref}
+      disabled={disabledProp ?? disabled}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<'button', IconButtonProps>;
+
+// ----------------------------------------------------------------------------
+
+/**
+ * SearchBox.Button component to add to your SearchBox.
+ */
+export type SearchBoxCloseButtonProps = PolymorphicComponentProps<
+  'button',
+  IconButtonProps
+>;
+
+const SearchBoxCloseButton = React.forwardRef((props, ref) => {
+  const {
+    children,
+    onClick: onClickProp,
+    size,
+    disabled: disabledProp,
+    ...rest
+  } = props;
+
+  const {
+    onClose,
+    size: sizeContext,
+    disabled,
+  } = useSafeContext(SearchBoxContext);
+
+  return (
+    <SearchBoxButton
+      ref={ref}
+      aria-label='Close searchbox'
+      size={size ?? sizeContext}
+      disabled={disabledProp ?? disabled}
+      onClick={(e) => {
+        onClickProp?.(e);
+        onClose?.();
```

Comment:
all of the event handlers need to check for `isDefaultPrevented`

Reply:
what do you think about making a utility function to combine event handlers like this?

```jsx
<button
  onClick={composeEventHandlers(onClickProp, onClose)}
/>
```
it will call the first argument, then check for `defaultPrevented`, then early return or call the other arguments

Reply:
That would be nice to have. However I don't want to change any other components in this PR. So IF I create this util function, I would only add it to SearchBox.

Reply:
lets do it in a separate PR then